src/Camelot.Services.Windows/WindowsShellIconsService.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Camelot.Services.Abstractions;
+using Camelot.Services.Abstractions.Models;
+using Camelot.Services.Windows.ShellIcons;
+using Camelot.Images;
+
+using SystemBitmap = System.Drawing.Bitmap;
+using AvaloniaBitmap = Avalonia.Media.Imaging.Bitmap;
+
+
+namespace Camelot.Services.Windows;
+
+
+[SupportedOSPlatform("windows")]
+public class WindowsShellIconsService : IShellIconsService
+{
+    public ImageModel GetIconForExtension(string extension)
+    {
+        if (string.IsNullOrEmpty(extension))
+            throw new ArgumentNullException(nameof(extension));
+        if (!extension.StartsWith("."))
+            throw new ArgumentOutOfRangeException(nameof(extension));
```

Comment:
`ArgumentException` is probably better because there is no range for it to be our of range